dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -449,16 +451,18 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
 				    GregorianCalendar expiration = new GregorianCalendar();
 					expiration.add(java.util.Calendar.MONTH, -1);
 					resp.setHeader("Expires", DownloadUtil.httpDate.get().format(expiration.getTime()));
-					resp.setHeader("Cache-Control", "max-age=-1");
+					if (!resp.containsHeader(HttpHeaders.CACHE_CONTROL)) {
+						resp.setHeader(HttpHeaders.CACHE_CONTROL, "max-age=-1");
+					}
 				}
 			}
 
 			String rangeHeader = req.getHeader("range");
 			if(UtilMethods.isSet(rangeHeader)){
 
-				try {
+				try (FileInputStream fileInputStream = new FileInputStream(data.getDataFile())) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)